dask_match/core.py

```diff
@@ -106,15 +106,15 @@ def __repr__(self):
         return str(self)
 
     def __getattr__(self, key):
-        if key in type(self)._parameters:
-            idx = type(self)._parameters.index(key)
-            return self.operands[idx]
-        elif key in dir(type(self)):
-            return object.__getattribute__(self, key)
-        elif key in self.columns:
-            return self[key]
-        else:
-            return object.__getattribute__(self, key)
```

Comment:
I'm curious, how did this come about?

Also, this seems unnecessarily nested.  I suspect that if this is necessary then this would be better with something like the following:

```python
if key == "__name__":
    return object.__getattribute__(self, key)
else:
    ...
```

I'll probably be pretty allergic to anything that increases nesting

Reply:
Like much of this PR, this change is just a temporary hack to get the existing tests running :)

The problem I ran into was that calling `self.columns` often requires us to actually parse parquet metadata to calculate `self._meta`, which doesn't work when `"myfile.parquet"` is not a real parquet dataset. Although this shouldn't "break" when the dataset is real, it also feels unnecessary to parse metadata when you are retrieving an attribute that doesn't require you to do so.